packages/core/http/core-http-router-server-internal/src/router.ts

```diff
@@ -119,6 +125,47 @@ function validOptions(
   return { ...options, body };
 }
 
+/**
+ * Adds ELU timings for the executed function to the current's context transaction
+ *
+ * @param callback The callback to instrument
+ * @param path The request path
+ */
+async function addEluTimings<T>(callback: () => Promise<T>, path: string, log: Logger): Promise<T> {
+  const startUtilization = performance.eventLoopUtilization();
+  const start = performance.now();
+
+  const response = await callback();
+
+  const { active, utilization } = performance.eventLoopUtilization(startUtilization);
+
+  apm.currentTransaction?.addLabels({
+    event_loop_utilization: utilization,
+    event_loop_active: active,
+  });
+
+  const duration = performance.now() - start;
+
+  if (active > THRESHOLD_ELA && utilization > THRESHOLD_ELU) {
```

Comment:
nit: sanity check only: should it be OR?

Reply:
no, AND - if only one condition matches it's going to fire e.g. on long running requests that do relatively little CPU work (or are impacted by lots of other CPU work from other requests) or very short requests that relatively do a lot of CPU work but which in absolute terms is negligible.

Reply:
makes sense! thanks for confirming!